File: csharp/TranscribeFileShort/TranscribeFileShort.cs
using Soniox.Types;
using Soniox.Client;
using Soniox.Client.Proto;

using var client = new SpeechClient();

var completeResult = await client.TranscribeFileShort(
    "../../test_data/test_audio.flac",
    new TranscriptionConfig
    {
        Model = "en_v2",
    });

Result result = (completeResult as SingleResult)!.Result;

var text = string.Join("", result.Words.Select(word => word.Text).ToArray());
Console.WriteLine("Text: " + text);
Console.WriteLine("Tokens:");
foreach (var word in result.Words)
{
    Console.WriteLine($"    '{word.Text}' {word.StartMs} {word.DurationMs}");
}

File: speech_to_text_legacy/node/transcribe_file_short.js
const { SpeechClient } = require("@soniox/soniox-node");

/**
 * Set your Soniox Cloud API key:
 * from command line: export SONIOX_API_KEY=<YOUR-API-KEY>
 * or
 * pass config object: new SpeechClient({ api_key: "<YOUR-API-KEY>" })
 */
const speechClient = new SpeechClient();

(async function () {
    const result = await speechClient.transcribeFileShort(
        "../test_data/test_audio.flac",
        {
            model: "en_v2",
        }
    );

    console.log("Text: " + result.words.map((word) => word.text).join(""));
    console.log("Tokens:");
    for (const word of result.words) {
        console.log(`    '${word.text}' ${word.start_ms} ${word.duration_ms}`);
    }
})();
